sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1189,15 +1189,32 @@ def configure_tx_output_power(self, api, lport, tx_power):
            self.log_error("{} configured tx power {} > maximum power {} supported".format(lport, tx_power, max_p))
         return api.set_tx_power(tx_power)
 
-    def configure_laser_frequency(self, api, lport, freq, grid=75):
-        _, _,  _, lowf, highf = api.get_supported_freq_config()
+    def config_laser_frequency_validate(self, api, lport, freq, grid=75):
```

Comment:
@balram-csco please rename it to -> `validate_frequency_and_grid()`

Reply:
renamed function to validate_frequency_and_grid()